Source/Core/Core/Boot/Boot.cpp

```diff
@@ -475,5 +475,19 @@ bool CBoot::BootUp()
 	// Not part of the binary itself, but either we or Gecko OS might insert
 	// this, and it doesn't clear the icache properly.
 	HLE::Patch(0x800018a8, "GeckoCodehandler");
+
+	u32 argvMagic = Memory::Read_U32(PC + 4);
+	if (argvMagic == 0x5f617267) {
+		std::string commandLine;
+		for (const auto& arg : _StartupPara.m_args) {
+			commandLine.append(arg);
+			commandLine.push_back('\0');
+		}
+		u32 argvLocation = 0x80003200; // TODO: Where should this actually be?
```

Comment:
The HBC behavior here is to use the page just before the loader area for argv data. So, 0x8132F000 to 0x81330000. Make sure something limits the size of commandLine to 1024 -- in case some crazy homebrew relies on overwriting the loader area with their own data.